pkgs/by-name/ne/nexusmods-app/package.nix

```diff
@@ -10,11 +10,10 @@
   libICE,
   libSM,
   libX11,
-  nexusmods-app,
   runCommand,
   pname ? "nexusmods-app",
 }:
-buildDotnetModule rec {
+buildDotnetModule (finalAttrs: rec {
   inherit pname;
   version = "0.4.1";
```

Comment:
Would it be better to set `version` in a let-block rather than use `rec`?

I think with nixfmt it could end up kinda messy having a let block just for the one variable though.

```nix
  pname ? "nexusmods-app",
}:
let
  version = "0.4.1";
in
buildDotnetModule (finalAttrs: {
  inherit pname version;

```

I'm not sure whether the update script can bump the version when defined like this, either.

Reply:
I personally like using simple syntax like `let + in` rather than advanced features like `finalAttrs` or especially `rec` to achieve this. But that's just, like, my opinion, man.

Reply:
I wouldn't call `rec` an advanced feature, but I agree it can be less obvious than reading a `let-in` block.

As for `finalAttrs`, that serves a different purpose to the other two; namely it makes `overrideAttrs` work correctly.

Reply:
This is fine like it is.